File: backend/src/main/java/org/ilisi/backend/model/Review.java
package org.ilisi.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private String id;

    @ManyToOne
    @JoinColumn(name = "student_id", referencedColumnName = "id")
    private Student student;

    @ManyToOne
    @JoinColumn(name = "institut_id", referencedColumnName = "id")
    @JsonIgnore
    private Institut institut;

    private String review;
    private int rating;

    @CreationTimestamp
    @Column(updatable = false, nullable = false)
    private Instant createdOn;

    @UpdateTimestamp
    private Instant lastUpdatedOn;
}

File: frontend/src/app/models/review.ts
import Student from "./student";

class Review {
    id?: string;
    review: string;
    rating: number;
    createdOn?: Date;
    institut: any;
    student?: Student;

  constructor(
    review: string,
    rating: number,
    institut: any,
    id?: string,
    createdOn?: Date,
    student?: Student
  ) {
    this.id = id;
    this.review = review;
    this.rating = rating;
    this.createdOn = createdOn;
    this.institut = institut;
    this.student = student;
  }
}

export default Review;